tests/storage/storage.go

```diff
@@ -965,14 +964,19 @@ var _ = SIGDescribe("Storage", func() {
 						}
 						return k8sv1.ConditionFalse
 					}, 30, 1).Should(Equal(k8sv1.ConditionTrue))
+					pod, err = ThisPod(pod)()
+					Expect(err).ToNot(HaveOccurred())
 
 					By("Determining the size of the mounted directory")
 					diskSizeStr, _, err := tests.ExecuteCommandOnPodV2(virtClient, pod, pod.Spec.Containers[0].Name, []string{"/usr/bin/bash", "-c", fmt.Sprintf("df %s | tail -n 1 | awk '{print $4}'", mountDir)})
 					Expect(err).ToNot(HaveOccurred())
 					diskSize, err = strconv.Atoi(strings.TrimSpace(diskSizeStr))
 					diskSize = diskSize * 1000 // byte to kilobyte
 					Expect(err).ToNot(HaveOccurred())
+				})
 
+				AfterEach(func() {
+					tests.RemoveHostDiskImage(diskPath, pod.Spec.NodeName)
```

Comment:
It looks like part of the cleanup (umount) happens on the pod PreStop condition.
Perhaps it'd be better to add this additional cleanup there so there's no ambiguity about the order in which things happen.
(I'm assuming this works because the umount happens afterwards)

Reply:
It seems that the `PreStop` hook was not called at all and the directory never got umounted. I had to add an explicit deletion of the hostpath pod in order to trigger that. Should be fine now.